packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -759,15 +764,29 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 	 * - close emits "closed"
 	 * - close cannot be called after dispose
 	 */
-	public close(error?: ICriticalContainerError): void {
+	public close(
+		errorOrReason?: ICriticalContainerError | DisconnectReason,
+		error?: ICriticalContainerError,
+	): void {
 		if (this._closed) {
 			return;
 		}
 		this._closed = true;
 
-		this.connectionManager.dispose(error, true /* switchToReadonly */);
+		let finalError: ICriticalContainerError | undefined;
+		let disconnectReason: DisconnectReason | undefined;
+
+		if (typeof errorOrReason === "string") {
```

Comment:
Is there anything we can do here to express this in terms of `DisconnectReason`, to make things match more obviously? At runtime it can only be a string so I get the check, but looking at the code it might not be as obvious.

Reply:
I have created isDisconnectReason; https://github.com/microsoft/FluidFramework/pull/23550/files#diff-1ccc57c98de9071338250c829ad175908391e8f6f3add01fdcb906ff9507a169R33